Core/src/org/sleuthkit/autopsy/casemodule/UnpackagePortableCaseProgressDialog.java

```diff
@@ -142,65 +149,89 @@ public void propertyChange(PropertyChangeEvent evt) {
         @NbBundle.Messages({
             "UnpackageWorker.doInBackground.errorFinding7zip=Could not locate 7-Zip executable",
             "UnpackageWorker.doInBackground.errorCompressingCase=Error unpackaging case",
-            "UnpackageWorker.doInBackground.canceled=Unpackaging canceled by user",})
+            "UnpackageWorker.doInBackground.canceled=Unpackaging canceled by user",
+            "UnpackageWorker.doInBackground.previousSeenCase=Case {0} has been previously opened do you want to open it again?"})
         @Override
         protected Void doInBackground() throws Exception {
 
-            // Find 7-Zip
-            File sevenZipExe = locate7ZipExecutable();
-            if (sevenZipExe == null) {
-                setDisplayError(Bundle.UnpackageWorker_doInBackground_errorFinding7zip());
-                throw new TskCoreException("Error finding 7-Zip executable"); // NON-NLS
-            }
-
-            String outputFolderSwitch = "-o" + outputFolder; // NON-NLS
-            ProcessBuilder procBuilder = new ProcessBuilder();
-            procBuilder.command(
-                    sevenZipExe.getAbsolutePath(),
-                    "x", // Extract
-                    packagedCase,
-                    outputFolderSwitch
-            );
-
-            try {
-                Process process = procBuilder.start();
-
-                while (process.isAlive()) {
-                    if (this.isCancelled()) {
-                        setDisplayError(Bundle.UnpackageWorker_doInBackground_canceled());
-                        return null;
+            // Check to see if this case has been already opened before
+            String caseUnpackedBefore = hasCaseBeenUnpackedBefore(packagedCase);
+            if (caseUnpackedBefore != null) {
+                if (MessageNotifyUtil.Message.confirm(Bundle.UnpackageWorker_doInBackground_previousSeenCase(packagedCase))) {
```

Comment:
Shouldn't it fall through to unzipping the portable case if I say no? Granted I haven't tested the code yet, I'm still only reading it. The if else block implies it will only open the previously seen case.

Reply:
Yes. Fixed